libs/api/domains/syslumenn/src/lib/client/syslumenn.client.ts

```diff
@@ -20,30 +26,57 @@ export class SyslumennClient {
     private clientConfig: SyslumennClientConfig,
   ) {}
 
-  private async getToken() {
+  private async login() {
     const config = {
       notandi: this.clientConfig.username,
       lykilord: this.clientConfig.password,
     }
 
     const response: { data: ILogin } = await this.httpService
-      .post(`${this.clientConfig.url}/dev/v1/Innskraning`, config)
+      .post(`${this.clientConfig.url}/v1/Innskraning`, config)
       .toPromise()
 
-    this.accessToken = response.data.audkenni
+    this.id = response.data.audkenni
+    this.accessToken = response.data.accessToken
   }
 
   async getHomestays(year?: number): Promise<IHomestay[] | null> {
-    await this.getToken()
+    await this.login()
 
     const url = year
-      ? `${this.clientConfig.url}/dev/v1/VirkarHeimagistingar/${this.accessToken}/${year}`
-      : `${this.clientConfig.url}/dev/v1/VirkarHeimagistingar/${this.accessToken}`
+      ? `${this.clientConfig.url}/v1/VirkarHeimagistingar/${this.id}/${year}`
+      : `${this.clientConfig.url}/v1/VirkarHeimagistingar/${this.id}`
 
     const response: { data: IHomestay[] } = await this.httpService
       .get(url)
       .toPromise()
 
     return response.data
   }
+
+  async uploadData(persons: Person[], attachment: Attachment): Promise<string> {
+    await this.login()
+
+    const url = `${this.clientConfig.url}/api/v1/SyslMottakaGogn`
```

Comment:
Should it not be like this?

```suggestion
    const url = `${this.clientConfig.url}/v1/SyslMottakaGogn`
```

Reply:
No but it should be ☹️ But as you can see here: https://api.syslumenn.is/dev/swagger/index.html#/ then for whatever reason then this endpoint has and extra `/api/` in it's path.

Reply:
Interesting :thinking: